tgui/packages/tgui/interfaces/common/sorting/SortableTable/utils.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { SortDirection } from "../type";
+import { SearchState, SortableTableHeaderConfig, SortableTableRowProps, SortState } from "./type";
+
+export const onSortClick = (index: number, current: SortState, setCallback: (nextState: SortState) => void) => {
+  if (current !== null) {
+    if (current.index === index) {
+      setCallback({
+        dir: (current.dir === SortDirection.Asc ? SortDirection.Desc : SortDirection.Asc),
+        index: index,
+      });
+    } else {
+      setCallback({ dir: SortDirection.Asc, index: index });
+    }
+  } else {
+    setCallback({ dir: SortDirection.Asc, index: index });
+  }
+};
+
+export const sortAndFilterRows = (
+  headerConfig: SortableTableHeaderConfig[],
+  rowConfig: SortableTableRowProps[],
+  sortState: SortState,
+  searchState: SearchState): SortableTableRowProps[] => {
+
+  const first = rowConfig[0];
+  let sorted = rowConfig;
+  if (headerConfig.length !== first.cells.length) {
+    throw new Error(`Inconsistent number of fields in table: ${headerConfig.length}x${first.cells.length}`);
+  }
```

Comment:
```suggestion
  const first = rowConfig[0];
  if (headerConfig.length !== first.cells.length) {
    throw new Error(`Inconsistent number of fields in table: ${headerConfig.length}x${first.cells.length}`);
  }
  let sorted = rowConfig;
```

Reply:
Also, make sure to handle the case where you have no contents in `rowConfig` (i.e. empty array). Having no data to sort/search is a valid state, and should not cause an error.